### PR TITLE
Voice and Client bug fixes

### PR DESCRIPTION
**Pull request by briantanner**
- Fix 'channels' of undefined in Client.getMessage
- Remove inconsistent promise rejection in leaveVoiceChannel
- Fix undefined pending guilds in readyHandler and disconnectHandler in VoiceConnectionManager

### CODE REVIEW

**Comment by briantanner** [comment]
The voice stuff may need looking at further

File: lib/Client.js [hunk 1]
```diff
@@ -276,7 +276,16 @@ class Client extends EventEmitter {
     * @returns {GuildChannel | GroupChannel | PrivateChannel}
     */
     getChannel(channelID) {
-        return this.channelGuildMap[channelID] ? this.guilds.get(this.channelGuildMap[channelID]).channels.get(channelID) : this.privateChannels.get(channelID) || this.groupChannels.get(channelID);
+        const guildID = this.channelGuildMap[channelID];
```
Comment (brussell98) [inline lib/Client.js]: This seems to be uneeded

**Review by brussell98 — CHANGES_REQUESTED** [review]

File: lib/Client.js (hunk 1 above)
Comment (briantanner) [reply lib/Client.js]: Ahh right, thanks for pointing it out

**Comment by abalabahaha** [comment]
Its probably better to figure out why channelGuildMap has data for guilds that don't appear to exist, rather than checking for null/undefined there

**Pull request merged**